storage/src/main/java/tech/pegasys/artemis/storage/CombinedChainDataClient.java

```diff
@@ -178,4 +185,49 @@ private boolean isFinalized(final UnsignedLong slot) {
     }
     return Optional.ofNullable(recentChainData.getBestBlockRoot());
   }
+
+  public boolean isStoreAvailable() {
+    return (recentChainData.getStore() != null);
+  }
+
+  /**
+   * Gets a list of CommitteeAssignments for a specified epoch.
+   *
+   * @param epoch - the current or historic epoch
+   * @return list of CommitteeAssignments
+   */
+  public SafeFuture<List<CommitteeAssignment>> getCommitteeAssignmentAtEpoch(UnsignedLong epoch) {
+    final UnsignedLong startingSlot = compute_start_slot_at_epoch(epoch);
+
+    SafeFuture<Optional<BeaconState>> future =
+        getStateAtSlot(startingSlot, recentChainData.getBestBlockRoot());
```

Comment:
Committees are always calculated for one epoch ahead.  So if you want to calculate committees for `epoch` you need to get the state for `epoch - 1`.  The exception is epoch 0 which uses genesisState (as does epoch 1).

So you are actually able to calculate committees up to one epoch ahead of the current one.

Reply:
ok i've done that and now i think our testing may be inadequate as no tests broke, suggesting that there's too much mocking to see this pretty significant change.